docs/src/4-differentiate-efms.jl

```diff
@@ -0,0 +1,145 @@
+# # Differentiating the EFMs/OFMs of an optimal solution
+
+# The optimal flux distribution of any metabolic model can be written as a 
+# weighted sum of the EFMs of that model. We are interested in calculating 
+# the sensitivity of these weightings to the model parameters. 
+
+using DifferentiableMetabolism
+
+using ElementaryFluxModes
+using FastDifferentiation
+using COBREXA
+using Tulip 
+using ConstraintTrees
+using LinearAlgebra
+
+# ## Build a simple enzyme constrained model
+
+# The code used to construct the model is located in `test/simple_model.jl`, but
+# it is not shown here for brevity.
+
+include("../../test/efm_simple_model.jl"); #hide
+
+model
+
+parameter_values = Dict{Symbol, Float64}()
+reaction_isozymes = Dict{String,Dict{String,ParameterIsozyme}}() # a mapping from reaction IDs to isozyme IDs to isozyme structs.
+float_reaction_isozymes = Dict{String,Dict{String,COBREXA.Isozyme}}() #src
+for rid in AbstractFBCModels.reactions(model)
+    grrs = AbstractFBCModels.reaction_gene_association_dnf(model, rid)
+    isnothing(grrs) && continue # skip if no grr available
+    haskey(rid_kcat, rid) || continue # skip if no kcat data available
+    for (i, grr) in enumerate(grrs)
+
+        kcat = rid_kcat[rid]
+        parameter_values[Symbol(rid)] = kcat
+
+        d = get!(reaction_isozymes, rid, Dict{String,ParameterIsozyme}())
+        d["isozyme_$i"] = ParameterIsozyme(
+            gene_product_stoichiometry = Dict(grr .=> fill(1.0, size(grr))), # assume subunit stoichiometry of 1 for all isozymes
+            kcat_forward = FastDifferentiation.Node(Symbol(rid)),
+            kcat_reverse = 0.0,
+        )
+        d2 = get!(float_reaction_isozymes, rid, Dict{String,COBREXA.Isozyme}()) #src
+        d2["isozyme_$i"] = COBREXA.Isozyme( #src
+            gene_product_stoichiometry = Dict(grr .=> fill(1.0, size(grr))), #src
+            kcat_forward = kcat, #src
+            kcat_reverse = 0.0, #src
+        ) #src
+    end
+end
+
+
+km = build_kinetic_model(
+    model;
+    reaction_isozymes,
+    gene_product_molar_masses,
+    capacity,
+)
+
+ec_solution, _, _, _ = optimized_constraints_with_parameters(
+    km,
+    parameter_values;
+    objective = km.objective.value,
+    optimizer = Tulip.Optimizer,
+    modifications = [COBREXA.set_optimizer_attribute("IPM_IterationsLimit", 10_000)],
+)
+
+ec_solution.fluxes
+
+ec_solution_cobrexa = enzyme_constrained_flux_balance_analysis( #src
```

Comment:
maybe call this `ec_solution_fba`, cobrexa can produce whatever else things too

Reply:
will do